### PR TITLE
skip affinity check when kubelet admit pods

### LINKED ISSUE #99708: Pod status becomes MatchNodeSelector after restart kubelet
#### What happened:

step 1: label node0 with region=r0
step 2: schedule a pod A with labelSelector region=r0 to node0, and pod A up and running
step 3: label node0 with region=rx
step 4: restart kubelet on node0, after kubelet restart, pod A evicted by kubelet and status becomes MatchNodeSelector

#### What you expected to happen:

pod A up and running on node0 even after we change nodeLabel and restart kubelet

#### How to reproduce it (as minimally and precisely as possible):

see `What happened`

#### Environment:
- Kubernetes version (use `kubectl version`): v1.14.4
- Cloud provider or hardware configuration: baremetal
- OS (e.g: `cat /etc/os-release`): -
- Kernel (e.g. `uname -a`): -
- Install tools: -
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by weiwei04**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Skip pod NodeAffinity when kubelet admit pods

#### Which issue(s) this PR fixes:

Fixes #99708 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/lifecycle/predicate.go [hunk 1]
```diff
@@ -236,9 +234,6 @@ func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]Pr
 		})
 	}
 
-	if !pluginhelper.PodMatchesNodeSelectorAndAffinityTerms(pod, nodeInfo.Node()) {
```
Comment (Pingan2017) [inline pkg/kubelet/lifecycle/predicate.go:239]: Maybe we need to keep the nodeselector check.

Comment (weiwei04) [reply pkg/kubelet/lifecycle/predicate.go:239]: according to https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity, nodeSelector is like requiredDuringSchedulingIgnoredDuringExecution

**Review by ehashman — COMMENTED** [review]
/remove-priority important-soon
/priority important-longterm

This implementation seems incorrect to me. We're removing a test and a bunch of code and not replacing it with anything.
/hold

**Pull request closed without merging**